tests/e2e/e2e_test.go

```diff
@@ -86,7 +91,40 @@ func (s *IntegrationTestSuite) TestSlashing() {
 	s.testSlashing(chainAPI)
 }
 
-// todo add fee test with wrong denom order
+func (s *IntegrationTestSuite) TestFeeWithWrongDenomOrder() {
+	chainAPI := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	
+	// Get the first validator's address
+	val1 := s.chainA.validators[0]
+	valAddr := val1.keyInfo.GetAddress()
+	
+	// Create a transaction with fees in wrong denom order
+	fees := sdk.NewCoins(
+		sdk.NewCoin("stake", sdk.NewInt(1000)),
+		sdk.NewCoin("atom", sdk.NewInt(100)),
+	)
+	
+	msg := banktypes.NewMsgSend(
+		valAddr,
+		valAddr,
+		sdk.NewCoins(sdk.NewCoin("atom", sdk.NewInt(1))),
+	)
+	
+	// Build and sign the transaction
+	txBuilder := s.chainA.clientCtx.TxConfig.NewTxBuilder()
+	err := txBuilder.SetMsgs(msg)
+	s.Require().NoError(err)
+	
+	txBuilder.SetFeeAmount(fees)
+	txBuilder.SetGasLimit(200000)
+	
+	// The transaction should still be accepted and processed correctly
+	// despite the fees being in a different order
+	tx, err := s.chainA.SignAndBroadcastTx(val1.keyInfo, txBuilder)
+	s.Require().NoError(err)
+	s.Require().Equal(uint32(0), tx.Code)
```

Comment:
Would be good to use a `require.Eventually` block like in https://github.com/cosmos/gaia/blob/accd9978e2f9fbe8ae296dd40d24e7c4617b7f9c/tests/e2e/e2e_bank_test.go#L40 to actually validate the send worked.